models/migrations/v109.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"crypto/md5"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+
+	"xorm.io/xorm"
+)
+
+func renameExistingUserAvatarName(x *xorm.Engine) error {
+	sess := x.NewSession()
+	defer sess.Close()
+
+	type User struct {
+		ID        int64  `xorm:"pk autoincr"`
+		LowerName string `xorm:"UNIQUE NOT NULL"`
+		Avatar    string
+	}
+	deleteList := make(map[string]struct{})
```

Comment:
```suggestion
	var deleteList []string
```
why not use normal string slice?

Reply:
At first I used normal string slice but then I replaced it with map.

Several users can have same avatar name*, so if we use normal slice for storing this, duplicate entries will be stored. This will cause problems in later parts of the code.


\* Resolving the avatar name duplication is the main reason of this pr.

Reply:
@6543, could you please provide any further feedback regarding my response ?

Reply:
if you add a new "string" you could use `util.IsStringInSlice("string", slice)` when adding a new string - but map already do the job for you :+1:

Reply:
would say resolved

Reply:
:+1: